Shell/FOOLElimination.cpp

```diff
@@ -149,13 +150,11 @@ FormulaUnit* FOOLElimination::apply(FormulaUnit* unit) {
     return rectifiedUnit;
   }
 
-  /*
-   * MS/TODO: We should be presenting the new formula as following
-   * from the rectifiedUnit and the generated definitions
-   * (similarly to how this is done with Naming)
-   */
+  // add the master premise to the definitions and pass them to the inference object
+  UnitList::push(rectifiedUnit,_currentDefs);
   FormulaUnit* processedUnit = new FormulaUnit(processedFormula,
-      NonspecificInference1(InferenceRule::FOOL_ELIMINATION, rectifiedUnit));
+      NonspecificInferenceMany(InferenceRule::FOOL_ELIMINATION, _currentDefs));
+  _currentDefs = UnitList::empty();
 
```

Comment:
Remind me - does `processedUnit` now own the previous value of `_currentDefs`, or are we leaking it here?

Reply:
According to my shallow investigation `Inference::destroy` takes care of it eventually, assuming the inference itself is ever destroyed.

Reply:
It's potentially leaking (as many other things in vampire), but it's meant to be owned by the inference. 

Note that just before _currentDefs are passed to the inference, the original formula is added as well.